test/Microsoft.ML.Tests/DatabaseLoaderTests.cs

```diff
@@ -112,258 +153,48 @@ public void IrisSdcaMaximumEntropy()
-    }
-
-    internal sealed class MockConnection : DbConnection
-    {
-        private string _dataPath;
-        private TextLoader _textLoader;
-
-        public MockConnection(MLContext context, DatabaseLoader databaseLoader)
-        {
-            var outputSchema = databaseLoader.GetOutputSchema();
-            var readerColumns = new TextLoader.Column[outputSchema.Count];
-
-            for (int i = 0; i < outputSchema.Count; i++)
-            {
-                var column = outputSchema[i];
-                var columnType = column.Type.RawType;
-
-                Assert.True(columnType.TryGetDataKind(out var internalDataKind));
-                readerColumns[i] = new TextLoader.Column(column.Name, internalDataKind.ToDataKind(), i);
-            }
-
-            _textLoader = context.Data.CreateTextLoader(readerColumns);
-        }
-
-        public override string ConnectionString
-        {
-            get
-            {
-                return _dataPath;
-            }
-
-            set
-            {
-                _dataPath = value;
-            }
-        }
-
-        public override string Database => throw new NotImplementedException();
-
-        public override string DataSource => throw new NotImplementedException();
-
-        public IDataView DataView { get; private set; }
-
-        public override string ServerVersion => throw new NotImplementedException();
-
-        public override ConnectionState State => throw new NotImplementedException();
-
-        public override void ChangeDatabase(string databaseName) => throw new NotImplementedException();
-
-        public override void Close() => throw new NotImplementedException();
-
-        public override void Open()
-        {
-            DataView = _textLoader.Load(_dataPath);
-        }
-
-        protected override DbTransaction BeginDbTransaction(IsolationLevel isolationLevel) => throw new NotImplementedException();
-
-        protected override DbCommand CreateDbCommand() => new MockCommand(this);
-    }
-
-    internal sealed class MockCommand : DbCommand
-    {
-        public MockCommand(MockConnection connection)
+        private string GetTestDatabasePath(string databaseName)
         {
-            CommandText = string.Empty;
-            Connection = connection;
+            return Path.GetFullPath(Path.Combine("TestDatabases", $"{databaseName}.mdf"));
         }
 
-        public override string CommandText { get; set; }
-
-        public override int CommandTimeout
+        private string GetConnectionString(string databaseName)
         {
-            get => throw new NotImplementedException();
-            set => throw new NotImplementedException();
+            var databaseFile = GetTestDatabasePath(databaseName);
+            return $@"Data Source=(LocalDB)\MSSQLLocalDB;AttachDbFilename={databaseFile};Database={databaseName};Integrated Security=True";
```

Comment:
Your CI is failing with

```
System.PlatformNotSupportedException : LocalDB is not supported on this platform.
```

Reply:
I'll have to disable for Linux for the time being and can log an issue for adding a mysql database as well?

Reply:
It is also failing for MacOS. It would be unfortunate to not have any unix tests for the DatabaseLoader....

---
In reply to: [319267186](https://github.com/dotnet/machinelearning/pull/4138#discussion_r319267186) [](ancestors = 319267186)

Reply:
Right, but I'm not sure I'll get a MySQL database together before this needs to be in for the next preview.

When is that cutoff?